nuxeo-features/rest-api/nuxeo-rest-api-test/src/test/java/org/nuxeo/ecm/restapi/test/ConfigTest.java

```diff
@@ -82,20 +84,29 @@ public void itCanRetrieveSchemaDefinitionWithFieldConstraints() throws IOExcepti
         assertEquals("schema", node.get("entity-type").getTextValue());
         JsonNode fields = node.get("fields");
         assertTrue(fields.size() > 0);
-        JsonNode creatorField = fields.get("creator");
-        assertEquals("string", creatorField.get("type").getTextValue());
-        ArrayNode constraints = (ArrayNode) creatorField.get("constraints");
+
+        // Test that nature has a constraint checking if value exists in directory
+        JsonNode natureField = fields.get("nature");
+        assertEquals("string", natureField.get("type").getTextValue());
+        ArrayNode constraints = (ArrayNode) natureField.get("constraints");
         assertEquals(2, constraints.size());
-        JsonNode userConstraint = null;
+        JsonNode natureConstraint = null;
         for (JsonNode constraint : constraints) {
-            if ("userManagerResolver".equals(constraint.get("name").getTextValue())) {
-                userConstraint = constraint;
+            if ("directoryResolver".equals(constraint.get("name").getTextValue())) {
+                natureConstraint = constraint;
                 break;
             }
         }
-        assertNotNull(userConstraint);
-        JsonNode userConstraintParams = userConstraint.get("parameters");
-        assertEquals(2, userConstraintParams.size());
+        assertNotNull(natureConstraint);
+        JsonNode natureConstraintParams = natureConstraint.get("parameters");
+        assertEquals(1, natureConstraintParams.size());
+        assertEquals("nature", natureConstraintParams.get("directory").getTextValue());
+
+        // Test that creator doesn't have a constraint checking if user exists - it still has a constraint for type
+        JsonNode creatorField = fields.get("creator");
+        assertEquals("string", creatorField.get("type").getTextValue());
+        constraints = (ArrayNode) creatorField.get("constraints");
+        assertEquals(1, constraints.size());
```

Comment:
Privilege the assert methods message parameters to hold such an information:
`assertEquals("Creator must only have a constraint for type", 1, constraints.size());`